### PR TITLE
Refactor common broker validations

### PR DESCRIPTION
**Pull request by eriknelson**
Refactors common broker validations for reuse. NOOP.

### CODE REVIEW

File: pkg/apis/servicecatalog/validation/broker.go [hunk 1]
```diff
@@ -91,12 +87,30 @@ func validateClusterServiceBrokerSpec(spec *sc.ClusterServiceBrokerSpec, fldPath
...
+
+	if len(commonErrs) != 0 {
+		allErrs = append(commonErrs)
+	}
+
+	return allErrs
+}
+
+func validateCommonServiceBrokerSpec(spec *sc.CommonServiceBrokerSpec, fldPath *field.Path) field.ErrorList {
+	commonErrs := field.ErrorList{}
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/validation/broker.go]: why rename?

File: pkg/apis/servicecatalog/validation/broker.go [hunk 2]
```diff
@@ -91,12 +87,30 @@ func validateClusterServiceBrokerSpec(spec *sc.ClusterServiceBrokerSpec, fldPath
 		}
 	}
 
+	commonErrs := validateCommonServiceBrokerSpec(&spec.CommonServiceBrokerSpec, fldPath)
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/validation/broker.go]: why create a var and an if? I think append works fine on empty slices

Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/broker.go]: I think the assignment and if can be combined in one line if we keep this format.

**Comment by MHBauer** [comment]
weird, not sure why this was on my review list.

File: pkg/apis/servicecatalog/validation/broker.go (hunk 2 above)
Comment (n3wscott) [reply pkg/apis/servicecatalog/validation/broker.go]: true.

File: pkg/apis/servicecatalog/validation/broker.go (hunk 1 above)
Comment (n3wscott) [reply pkg/apis/servicecatalog/validation/broker.go]: it matches the naming of the spec object being validated.

**Pull request merged**